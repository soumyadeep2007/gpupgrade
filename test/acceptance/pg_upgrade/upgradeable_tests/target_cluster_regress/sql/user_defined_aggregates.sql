-- Copyright (c) 2017-2021 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------
CREATE TABLE table_used_in_sum_plus_42(a int);
INSERT INTO table_used_in_sum_plus_42 SELECT 1 FROM generate_series(1,10);

SELECT sum_plus_42(a) FROM table_used_in_sum_plus_42;

SELECT attrelid::regclass, array_accum(attname)
    FROM pg_attribute
    WHERE attnum > 0 AND attrelid = 'pg_tablespace'::regclass
    GROUP BY attrelid;

SELECT attrelid::regclass, array_accum(atttypid::regtype)
    FROM pg_attribute
    WHERE attnum > 0 AND attrelid = 'pg_tablespace'::regclass
    GROUP BY attrelid;
