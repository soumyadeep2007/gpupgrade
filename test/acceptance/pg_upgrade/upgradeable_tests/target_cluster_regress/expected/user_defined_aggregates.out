-- Copyright (c) 2017-2021 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------
CREATE TABLE table_used_in_sum_plus_42(a int);
CREATE
INSERT INTO table_used_in_sum_plus_42 SELECT 1 FROM generate_series(1,10);
INSERT 10

SELECT sum_plus_42(a) FROM table_used_in_sum_plus_42;
 sum_plus_42 
-------------
 52          
(1 row)

SELECT attrelid::regclass, array_accum(attname) FROM pg_attribute WHERE attnum > 0 AND attrelid = 'pg_tablespace'::regclass GROUP BY attrelid;
 attrelid      | array_accum                          
---------------+--------------------------------------
 pg_tablespace | {spcname,spcowner,spcacl,spcoptions} 
(1 row)

SELECT attrelid::regclass, array_accum(atttypid::regtype) FROM pg_attribute WHERE attnum > 0 AND attrelid = 'pg_tablespace'::regclass GROUP BY attrelid;
 attrelid      | array_accum                 
---------------+-----------------------------
 pg_tablespace | {name,oid,aclitem[],text[]} 
(1 row)
